mesonbuild/cmake/traceparser.py

```diff
@@ -396,6 +404,55 @@ def _cmake_set_target_properties(self, tline: CMakeTraceLine) -> None:
 
                 self.targets[i].properties[name] = value
 
+    def _cmake_add_dependencies(self, tline: CMakeTraceLine) -> None:
+        # DOC: https://cmake.org/cmake/help/latest/command/add_dependencies.html
+        args = list(tline.args)
+
+        if len(args) < 2:
+            return self._gen_exception('add_dependencies', 'takes at least 2 arguments', tline)
+
+        target = self.targets.get(args[0])
+        if not target:
+            return self._gen_exception('add_dependencies', 'target not found', tline)
+
+        target.depends += args[1:]
+
+    def _cmake_install(self, tline: CMakeTraceLine) -> None:
+        # DOC: https://cmake.org/cmake/help/latest/command/install.html
+        args = list(tline.args)
+
+        if len(args) < 2:
+            return self._gen_exception('install', 'takes at least 2 arguments', tline)
+
+        fn = None
+        install_targets = []
+        install_dir = None
+        def handle_target(arg: str):
+            for i in arg.split(';'):
+                target = self.targets.get(i)
+                if not target:
+                    return self._gen_exception('install', 'target not found', tline)
+                target.install = True
+                install_targets.append(target)
+
+        def handle_destination(arg: str):
+            install_dir = arg
```

Comment:
__[Flake8]__ [F841] local variable 'install_dir' is assigned to but never used ([link](https://sider.review/gh/repos/19784232/pulls/5989#issue-4637209))

<sub>posted by Sider</sub>